enterprise/signals.py

```diff
@@ -347,6 +350,35 @@ def create_enterprise_enrollment_receiver(sender, instance, **kwargs):     # pyl
         )
 
 
-# Don't connect this receiver if we dont have access to CourseEnrollment model
+def delete_enterprise_enrollment_receiver(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receives the ``UNENROLL_DONE`` signal, defined in
+    https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/signals/signals.py
+
+    and deletes the ``EnterpriseCourseEnrollment`` associated with the modified
+    ``student.CourseEnrollment``.
+    """
+    course_enrollment = kwargs['course_enrollment']
+
+    try:
+        user_id = course_enrollment.user.id
+        course_id = course_enrollment.course_id
+        enterprise_customer_user = EnterpriseCustomerUser.objects.get(user_id=user_id)
```

Comment:
`EnterpriseCustomerUser.objects.get(user_id=user_id)` assumes a user will only be linked to a single enterprise. If a user is linked to more than 1 enterprise, there will be several `EnterpriseCustomerUser` records for the specified user id.

Could we retrieve the `EnterpriseCourseEnrollment` records without needing to rely on a specific `EnterpriseCustomerUser` record?

```python
enterprise_course_enrollments = EnterpriseCourseEnrollment.objects.filter(
  enterprise_customer_user__user_id=user_id,
  course_id=course_id,
)
```

Reply:
Mmmmm, good point.

Reply:
Hmm, but what if they're enrolled in the same course in multiple enterprises, but only want to unenroll in their currently selected enterprise?

Reply:
@iloveagent57 If they're enrolled in the same course through multiple enterprises, wouldn't the user still have a single non-Enterprise (aka "normal") course enrollment record? As in, if that user goes to the B2C dashboard, they wouldn't see the course listed twice, correct?

If that assumption is true and the user unenrolls from that course, I feel like I'd expect the unenrollment to apply to both enterprise course enrollments.

If we only unenrolled for their currently selected enterprise, would we get into a situation where the user lost their actual course enrollment but we'd still have a `EnterpriseCourseEnrollment` record for the now-unenrolled course?